pkg/providers/instancetype/offering/provider.go

```diff
@@ -0,0 +1,274 @@
+			Offerings:    offerings,
+			Capacity:     it.Capacity,
+			Overhead:     it.Overhead,
+		})
+	}
+	return its
+}
+
+//nolint:gocyclo
+func (p *DefaultProvider) createOfferings(
+	it *cloudprovider.InstanceType,
+	nodeClass *v1.EC2NodeClass,
+	allZones sets.Set[string],
+	subnetZones map[string]string,
+	snapshotUUID *types.UID,
+) cloudprovider.Offerings {
+	itZones := sets.New(it.Requirements.Get(corev1.LabelTopologyZone).Values()...)
+
+	offerings := []cloudprovider.Offering{}
+	for zone := range allZones {
+		for _, capacityType := range it.Requirements.Get(karpv1.CapacityTypeLabelKey).Values() {
+			// Reserved capacity types are constructed separately, skip them for now.
+			if capacityType == karpv1.CapacityTypeReserved {
+				continue
+			}
+
+			isUnavailable := p.unavailableOfferings.IsUnavailable(it.Name, zone, capacityType)
+			_, hasSubnetZone := subnetZones[zone]
+			var price float64
+			var hasPrice bool
+			switch capacityType {
+			case karpv1.CapacityTypeOnDemand:
+				price, hasPrice = p.pricingProvider.OnDemandPrice(ec2types.InstanceType(it.Name))
+			case karpv1.CapacityTypeSpot:
+				price, hasPrice = p.pricingProvider.SpotPrice(ec2types.InstanceType(it.Name), zone)
+			default:
+				panic(fmt.Sprintf("invalid capacity type %q in requirements for instance type %q", capacityType, it.Name))
+			}
+			offering := cloudprovider.Offering{
+				Requirements: scheduling.NewRequirements(
+					scheduling.NewRequirement(karpv1.CapacityTypeLabelKey, corev1.NodeSelectorOpIn, capacityType),
+					scheduling.NewRequirement(corev1.LabelTopologyZone, corev1.NodeSelectorOpIn, zone),
+					scheduling.NewRequirement(v1.LabelCapacityReservationID, corev1.NodeSelectorOpDoesNotExist),
+				),
+				Price:     price,
+				Available: !isUnavailable && hasPrice && itZones.Has(zone) && hasSubnetZone,
+			}
+			if id, ok := subnetZones[zone]; ok {
+				offering.Requirements.Add(scheduling.NewRequirement(v1.LabelTopologyZoneID, corev1.NodeSelectorOpIn, id))
+			}
+			offerings = append(offerings, offering)
+		}
+	}
+
+	for i := range nodeClass.Status.CapacityReservations {
+		if nodeClass.Status.CapacityReservations[i].InstanceType != it.Name {
+			continue
+		}
+		reservation := &nodeClass.Status.CapacityReservations[i]
+
+		isUnavailable := p.unavailableOfferings.IsReservationUnavailable(reservation.ID)
+		_, hasSubnetZone := subnetZones[reservation.AvailabilityZone]
+		price := 0.0
+		if odPrice, ok := p.pricingProvider.OnDemandPrice(ec2types.InstanceType(it.Name)); ok {
+			// Divide the on-demand price by a sufficiently large constant. This allows us to treat the reservation as "free",
+			// while maintaining relative ordering for consolidation. If the pricing details are unavailable for whatever reason,
+			// still succeed to create the offering and leave the price at zero. This will break consolidation, but will allow
+			// users to utilize the instances they're already paying for.
+			price = odPrice / 100_000.0
```

Comment:
Did we validate that 100,000 is large enough? I think when I was looking at it, we needed to make sure that the most expensive OD was still cheaper than the cheapest spot and that brought me to a number that was around 1,000,000 -- you may want to go even a little higher to give yourself some buffer for larger, more expensive instance types to come out

Reply:
It's currently at 1,000,000 but we could 10x that to be safe.